nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -63,9 +63,13 @@ CollisionMonitor::on_configure(const rclcpp_lifecycle::State & /*state*/)
     return nav2_util::CallbackReturn::FAILURE;
   }
 
-  cmd_vel_in_sub_ = this->create_subscription<geometry_msgs::msg::Twist>(
-    cmd_vel_in_topic, 1,
-    std::bind(&CollisionMonitor::cmdVelInCallback, this, std::placeholders::_1));
+  cmd_vel_in_sub_ = std::make_shared<nav2_util::TwistSubscriber>(
```

Comment:
Same with velocity smoother: the output here also needs to be stamped and probably changes the way that this should be handled.

Here and in the velocity smoother: why not use the specific constructors for the specific stamped/unstamped parameter so we're not registering both cbs when unnecessary?

Reply:
Yep, good spot. I'll handle both with TwistSubscriber and TwistPublisher. I also modified the internal timestamping for the last command packet to use the incoming packet timestamp rather than node wall clock.

Reply:
50cc29245aafb83b29436de78060caec62170afb